src/server/object_services/object_server.js

```diff
@@ -162,7 +167,166 @@ async function delete_object_tagging(req) {
         version_id: info.version_id,
     };
 }
+////////////////////
+
+function calc_retention(req) {
+    dbg.log0('calc_retention:', req.rpc_params);
+    let obj_settings = req.rpc_params.lock_settings;
+
+    if (!obj_settings) {
+        const retention_conf = get_default_lock_config(req.bucket);
+        if (!retention_conf) return;
+
+        var today = new Date();
+        const retain_until_date = retention_conf.days ? new Date(today.setDate(today.getDate() + retention_conf.days)) :
+            new Date(today.setFullYear(today.getFullYear() + retention_conf.years));
+
+        obj_settings = { retention: { mode: retention_conf.mode, retain_until_date } };
+
+    } else if (obj_settings.retention && obj_settings.retention.retain_until_date) {
+        obj_settings.retention.retain_until_date = new Date(obj_settings.retention.retain_until_date);
+    }
+    return obj_settings;
+}
+
+function get_default_lock_config(bucket) {
+    dbg.log0('get_default_lock_config:', bucket.object_lock_configuration);
+    const bucket_info = bucket.object_lock_configuration;
+    if (bucket_info.object_lock_enabled !== 'Enabled') {
+        return;
+    }
+    if (bucket_info.rule) return bucket_info.rule.default_retention;
+}
+/**
+ * 
+ * get_object_legal_hold
+ *
+ */
+
+async function get_object_legal_hold(req) {
+    dbg.log0('get_object_legal_hold:', req.rpc_params);
+    load_bucket(req);
+    const obj = await find_object_md(req);
+    const info = get_object_info(obj, { 'role': req.role });
```

Comment:
no need for the quotes. but I think it's not needed at all. just a true/false variable

Reply:
removed the quotes, and regarding the true/false variable - same as above